### PR TITLE
refactor: terms of use

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

refactor the original `termsOfUse` object with `termsOfUseUrl` string.

image

## Testing

Test locally.

### CODE REVIEW

File: packages/core/src/routes/sign-in-experience.ts [hunk 1]
```diff
@@ -30,12 +30,19 @@ export default function signInExperiencesRoutes<T extends AuthedRouter>(router:
   router.patch(
     '/sign-in-exp',
     koaGuard({
-      body: SignInExperiences.createGuard.omit({ id: true }).partial(),
+      body: SignInExperiences.createGuard
+        .omit({ id: true, termsOfUseUrl: true })
+        .merge(
+          object({
+            termsOfUseUrl: string().url().optional().or(literal('')),
+          })
```
Comment (simeng-li) [inline packages/core/src/routes/sign-in-experience.ts:38]: Feels like the previous implementation are more clear.  Is there any specific reason to put it in koaGuard?

File: packages/schemas/alterations/next-1667896654-add-terms-of-use-url-column.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    const id = 'default';
+
+    await pool.query(sql`
+      alter table sign_in_experiences add column terms_of_use_url varchar(2048)
+    `);
+
+    const data = await pool.maybeOne<SignInExperience>(
```
Comment (simeng-li) [inline packages/schemas/alterations/next-1667896654-add-terms-of-use-url-column.ts:20]: what if there are multiple SignInExperience records?

File: packages/schemas/alterations/next-1667896654-add-terms-of-use-url-column.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    if (data) {
+      const {
+        termsOfUse: { enabled, contentUrl },
+      } = data;
+
+      if (enabled && contentUrl) {
+        await pool.query(sql`
+          update sign_in_experiences set terms_of_use_url = ${contentUrl} where id = ${id}
+        `);
```
Comment (simeng-li) [inline packages/schemas/alterations/next-1667896654-add-terms-of-use-url-column.ts:32]: Do we need to drop the original column? or mark it deprecated?

File: packages/schemas/alterations/next-1667896654-add-terms-of-use-url-column.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      }
+    }
+  },
+  down: async (pool) => {
+    await pool.query(sql`
+      alter table sign_in_experiences drop column terms_of_use_url
+    `);
```
Comment (simeng-li) [inline packages/schemas/alterations/next-1667896654-add-terms-of-use-url-column.ts:39]: noticed the original terms_of_use column is removed from the table schema in this PR.  If run-down script there will be not terms field left?

File: packages/ui/src/hooks/use-terms.ts [hunk 5]
```diff
@@ -12,12 +12,12 @@ const useTerms = () => {
...
   const { show } = useConfirmModal();
 
-  const { termsOfUse } = experienceSettings ?? {};
+  const { termsOfUseUrl } = experienceSettings ?? {};
 
   const termsOfUseIframeModalHandler = useCallback(async () => {
     const [result] = await show({
       className: styles.iframeModal,
-      ModalContent: () => createIframeConfirmModalContent(termsOfUse?.contentUrl),
+      ModalContent: () => createIframeConfirmModalContent(termsOfUseUrl ?? ''),
```
Comment (simeng-li) [inline packages/ui/src/hooks/use-terms.ts:20]: same here.

**Comment by xiaoyijun** [comment]
Convert to draft for this feature is not in the current cycle.

**Pull request closed without merging**